cpp/test/sg/rf_treelite_test.cu

```diff
@@ -205,21 +200,14 @@ class RfTreeliteTestCommon : public ::testing::TestWithParam<RfInputs<T>> {
     data_h.resize(data_len);
     inference_data_h.resize(inference_data_len);
 
-    for (int i = 0; i < params.n_rows; i++) {
-      for (int j = 0; j < params.n_cols; j++) {
-        float base = j * 100.0;
-        // Generate random numbers from 0.0 to 10.0
-        float random_noise =
-          static_cast<float>(rand()) / (static_cast<float>(RAND_MAX / 10.0));
-        // Populate data (assume Col major)
-        data_h[j * params.n_rows + i] = base + random_noise;
-        // Populate inference data (The same as data but in Row major)
-        inference_data_h[i * params.n_cols + j] = base + random_noise;
-      }
-    }
+    Random::Rng r(1234ULL);
+
+    r.uniform(data_d, data_len, T(0.0), T(100.0), stream);
+    r.uniform(inference_data_d, inference_data_len, T(0.0), T(100.0), stream);
 
-    updateDevice(data_d, data_h.data(), data_len, stream);
-    updateDevice(inference_data_d, inference_data_h.data(), data_len, stream);
+    updateHost(data_h.data(), data_d, data_len, stream);
+    updateHost(inference_data_h.data(), inference_data_d, inference_data_len,
+               stream);
```

Comment:
`cudaStreamSynchronize()` is needed after updateHost() before the host arrays can be used.